.npmrc

```diff
@@ -0,0 +1 @@
+package-lock=false
```

Comment:
Would it makes sense to include `package-lock.json` also in the `.gitignore`? Or is this automatically done by this as well?

Reply:
When configured like this, npm will not create a `package-lock.json` in the first place. So I thought it would be sufficient to do it like this (and even become visible in the git status if a `package-lock.json` does show up for some unexpected reason)...

Reply:
Ah, understood! I thought that it won't be uploaded when publishing the package, not that it would be avoided also when installing.

I'm in general not sure about not having the package-lock.json in place. As far as I can see this would imply that:
 - The CI would need to install from scratch the dependencies for every run.
 - There might be inconsistencies between developers about the dependency versions that are being used locally, as well as with the CI run.

Which advantages do you see for removing the `package-lock.json`?

Reply:
The thing is, everyone installing `@leanix/reporting-cli` over npm will get different package versions anyway, because npm will not include `package-lock.json` into the module (see https://unpkg.com/browse/@leanix/reporting-cli@1.0.0-beta.17/). So I'd argue that having a `package-lock.json` only creates an illusion of safety, where the CI keeps running green while actual installs for users get different dependencies that might be broken.

So the advantage of removing `package-lock.json` is to not have something that looks like but doesn't actually add safety. As far as I understand, the alternative is to [add an `npm-shrinkwrap.json`](https://docs.npmjs.com/cli/v6/configuring-npm/shrinkwrap-json) file, which does the same as `package-lock.json` but can be published. From the documentation, it reads like it is intended exactly for CLI modules. On the other hand I had a look at `pino-pretty` and `npm` itself, and both don't include an `npm-shrinkwrap.json`.

What do you think, should we try using `npm-shrinkwrap.json` instead?

Reply:
I see! Thank you for the explanation :) I wasn't fully aware about the complete picture.

I agree on your point that this gives us a false feeling of safety, since at the the user would install the package from scratch, and version dependencies would differ anyway. By removing the `package-lock.json` we would have the same problem, right? The user might have a different dependency tree than when the package was published. The difference now is that we experience this when developing the cli as well, right?

I'm totally fine with removing `package-lock.json`, since this has been already behaving like this since the first version, and as far as we know, there has not been any installation issues.

However, I would think, that there's not really a point to track security vulnerabilities due to dependencies, if, at the end, the user would have a different dependency constellation as in the repository.

Reply:
As said, I think, we could go for this solution. And if at any point in time, we get some installation issues of the package, we can think about `shrinkwrap` 👍